app/models/portfolio_item.rb

```diff
@@ -18,4 +18,9 @@ class PortfolioItem < ApplicationRecord
 
   belongs_to :portfolio
   validates :service_offering_ref, :presence => true
+
+  def resolved_workflow_ref
+    hierarchy_search = Catalog::HierarchicalModelSearch.new("workflow_ref", id, %w[PortfolioItem Portfolio])
+    hierarchy_search.process.field
```

Comment:
Realized we really want this to be `resolved_workflow_refs` and have it return an array of workflows.  For right now it will only return one.  In the near-future we could include a workflow assigned to the platform that the portforlio_item is associated to.  The Platform workflow would run parallel to the resolved portfolio_item workflow.

Changing this to an array now gets the code ready for that feature.

Reply:
Ok changed the logic to return an array of all of the results. Do we want to `compact!` the results so it only returns non-nil values or should we return them all that way one can map the array of models to the array of returned values? 

I can see the merit of both - maybe we could handle that on the calling side and not the service side.